core/style.js

```diff
@@ -2117,3 +2117,87 @@ CKEDITOR.tools.extend( CKEDITOR.editor.prototype, {
  * @cfg {String/Array/Boolean} [stylesSet='default']
  * @member CKEDITOR.config
  */
+
+/**
+ * Abstract class describing definition of a style.
+ *
+ * This virtual class illustrates the properties that developers can use to define and create
+ * style definitions.
+ *
+ * Style definition object represents a style as a set of properties describing CSS style rules and HTML attributes.
+ * It also store element type which can be used to provide valid HTML element as a wrapper or replacement for a selection's HTML.
+ * The {@link CKEDITOR.style} based on such definition can be applied to and removed from selection
+ * through various {@link CKEDITOR.style} methods.
+ *
+ *  	{
+ *			name: 'Special Title',
+ *			element: 'h1',
+ *			attributes: { class: 'my_class' },
+ *			styles: { color: 'red', 'font-size': '16px', 'font-width': 'bold' }
+ *		}
+ *
+ *	More on how editor content styles are handled can be found in the [Styles guide](#!/guide/dev_howtos_styles).
+ *
+ * @class CKEDITOR.style.definition
+ * @abstract
+ */
+
+/**
+ * Defines style type.
+ *
+ * There are three standard style types - {@link CKEDITOR#STYLE_INLINE}, {@link CKEDITOR#STYLE_BLOCK}
+ * or {@link CKEDITOR#STYLE_OBJECT}, each one related to the element used in the style rule and types of
+ * elements to which specific style can be applied.
+ *
+ * Plugins may define {@link CKEDITOR.style.customHandlers special style handlers} to customize style operations.
+ * To use special style handler, `type` property should be set as a name of the style handler, e.g. `widget`.
+ *
+ * More on style types can be found in the [Style Types section of the Styles guide](#!/guide/dev_styles-section-style-types).
+ *
+ * @property {String/Number} type=CKEDITOR.STYLE_INLINE
+ */
+
+/**
+ * An unique name assigned to a style definition. An name is used to differentiate style definitions e.g.
```

Comment:
Maybe something a little shorter like:

>  An unique style definition name. It could be used to differentiate definitions, like in Styles Combo plugin dropdown where it represents item labels.

?